File: src/types/network.ts
export type NetworkBackendKind = 'manual_lan' | 'radmin' | 'n2n';

export interface BackendSummary {
  id: NetworkBackendKind;
  name: string;
  installed: boolean;
  available: boolean;
  virtual_ip?: string | null;
  notes: string[];
}

export interface NetworkConfig {
  room_name?: string;
  secret?: string;
  supernode?: string;
  local_ip?: string;
}

export interface SetupResult {
  ok: boolean;
  message: string;
}

export interface BackendRuntimeStatus {
  backend_id: NetworkBackendKind;
  running: boolean;
  virtual_ip?: string | null;
  message: string;
}

export interface N2nDiagnostics {
  running: boolean;
  supernode_configured: boolean;
  supernode?: string;
  virtual_ip?: string;
  ack: boolean;
  pong: boolean;
  ok_link: boolean;
  auth_error: boolean;
  ip_mac_conflict: boolean;
  not_responding: boolean;
  tap_error?: boolean;
  last_error?: string | null;
  summary: string;
  log_path: string;
  recent_logs: string[];
  executable_found?: boolean;
  executable_path?: string | null;
  recorded_pid?: number | null;
  recorded_pid_running?: boolean;
  connection_state?: string;
  manual_start_command?: string | null;
}

export interface ConnectivityTarget {
  host: string;
  ports: number[];
  timeout_ms?: number;
  mode?: 'generic' | 'local_game_port' | 'n2n_game_port';
  protocol?: 'tcp' | 'udp' | 'tcp_udp';
}

export interface PortCheckResult {
  port: number;
  reachable: boolean;
  latency_ms?: number | null;
  error?: string | null;
}

export interface ConnectivityReport {
  target_host: string;
  reachable: boolean;
  latency_ms?: number | null;
  ports: PortCheckResult[];
  notes: string[];
}

File: src-tauri/src/models/network.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BackendSummary {
    pub id: String,
    pub name: String,
    pub installed: bool,
    pub available: bool,
    pub virtual_ip: Option<String>,
    pub notes: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct NetworkConfig {
    pub room_name: Option<String>,
    pub secret: Option<String>,
    pub supernode: Option<String>,
    pub local_ip: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SetupResult {
    pub ok: bool,
    pub message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BackendRuntimeStatus {
    pub backend_id: String,
    pub running: bool,
    pub virtual_ip: Option<String>,
    pub message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct N2nDiagnostics {
    pub running: bool,
    pub supernode_configured: bool,
    pub supernode: Option<String>,
    pub virtual_ip: Option<String>,
    pub ack: bool,
    pub pong: bool,
    pub ok_link: bool,
    pub auth_error: bool,
    pub ip_mac_conflict: bool,
    pub not_responding: bool,
    pub tap_error: bool,
    pub last_error: Option<String>,
    pub summary: String,
    pub log_path: String,
    pub recent_logs: Vec<String>,
    pub executable_found: bool,
    pub executable_path: Option<String>,
    pub recorded_pid: Option<u32>,
    pub recorded_pid_running: bool,
    pub connection_state: String,
    pub manual_start_command: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectivityTarget {
    pub host: String,
    pub ports: Vec<u16>,
    pub timeout_ms: Option<u64>,
    pub mode: Option<String>,
    pub protocol: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PortCheckResult {
    pub port: u16,
    pub reachable: bool,
    pub latency_ms: Option<u128>,
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectivityReport {
    pub target_host: String,
    pub reachable: bool,
    pub latency_ms: Option<u128>,
    pub ports: Vec<PortCheckResult>,
    pub notes: Vec<String>,
}
